### PR TITLE
uplift to v0.9.30

### PR DESCRIPTION
**Pull request by shunsukew**
Substrate uplift to `polkadot-v0.9.30` + Instant Finalization (solves https://github.com/AstarNetwork/swanky-node/issues/36)

keeping manual seal as it is (RPC to produce blocks and finalize), since it leaves the potential for features such as forwarding blocks and finalizing multiple blocks with one call by swanky-cli.

### CODE REVIEW

File: node/src/service.rs [hunk 1]
```diff
@@ -204,7 +205,7 @@ pub fn new_full(config: Configuration) -> Result<TaskManager, ServiceError> {
 		transaction_pool.clone().import_notification_stream().map(|_| {
 			sc_consensus_manual_seal::EngineCommand::SealNewBlock {
 				create_empty: false,
-				finalize: false,
+				finalize: true,
```
Comment (shunsukew) [inline node/src/service.rs:208]: Only this one line enables instant finalization

File: README.md [hunk 2]
```diff
@@ -3,19 +3,18 @@
...
+- [pallet-dapps-staking](https://github.com/AstarNetwork/astar-frame/tree/polkadot-v0.9.30/frame/dapps-staking) and ChainExtension to interact with it.
+- [pallet-assets](https://github.com/paritytech/substrate/tree/polkadot-v0.9.30/frame/assets).
+- [pallet-rmrk](https://github.com/AstarNetwork/rmrk-substrate/tree/polkadot-v0.9.30) (core, equip, market) and chain extensions for pallet-rmrk-core.
 
 It is optimized to local development purpose while removing unnecessary components such as P2P.
 More features and pallets to interact with (Contract <-> Runtime) will be added.
 
 ## Compatible ink! version
-ink! version `3.3.1` or lower is supported by pallet-contract polkadot-0.9.29 branch.
+ink! version `3.3.1` or lower is supported by pallet-contract polkadot-0.9.30 branch.
```
Comment (PierreOssun) [inline README.md:17]:
```suggestion
ink! version `3.4.0` or lower is supported by pallet-contract polkadot-0.9.30 branch.
```

**Review by PierreOssun — APPROVED** [review]
LGTM

**Comment by vikiival** [comment]
Thank you!

**Pull request merged**